pilot/proxy/envoy/policy.go

```diff
@@ -120,3 +125,128 @@ func applyClusterPolicy(cluster *Cluster,
 		}
 	}
 }
+
+func applyLoadBalancePolicy(cluster *Cluster, policy *routingv2.LoadBalancerSettings) {
+	if policy == nil || cluster.Type == ClusterTypeOriginalDST {
+		return
+	}
+
+	if consistent := policy.GetConsistentHash(); consistent != nil {
+		// cluster.LbType = LbTypeRingHash
+		// TODO: need to set special values in the route config
+		// see: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/load_balancing.html#ring-hash
+		log.Warn("Consistent hash load balancing type is not currently supported")
+	} else {
+		switch policy.GetSimple() {
+		case routingv2.LoadBalancerSettings_LEAST_CONN:
+			cluster.LbType = LbTypeLeastRequest
+		case routingv2.LoadBalancerSettings_RANDOM:
+			cluster.LbType = LbTypeRandom
+		case routingv2.LoadBalancerSettings_PASSTHROUGH:
+			cluster.LbType = LbTypeOriginalDST // FIXME: double-check
+		default:
+			cluster.LbType = LbTypeRoundRobin
+		}
+	}
+}
+
+func buildOutlierDetection(outlier *routingv2.OutlierDetection) *OutlierDetection {
+	if outlier != nil && outlier.Http != nil {
+		out := &OutlierDetection{
```

Comment:
I assume these are the default values from Envoy's config?

Reply:
yes, those are the defaults from the envoy docs and the istio proto's